src/test/java/hudson/plugins/mercurial/CachingCustomDirSCMTest.java

```diff
@@ -0,0 +1,53 @@
+package hudson.plugins.mercurial;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.regex.Pattern;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import hudson.model.FreeStyleProject;
+import hudson.tools.ToolProperty;
+
+public class CachingCustomDirSCMTest {
+
+	@Rule
+	public JenkinsRule j = new JenkinsRule();
+	@Rule
+	public MercurialRule m = new MercurialRule(j);
+	@Rule
+	public TemporaryFolder tmp = new TemporaryFolder();
+	private File repo;
+
+	private static final String CACHING_INSTALLATION = "caching-custom-dir";
+
+	@Before
+	public void setUp() throws Exception {
+		repo = tmp.getRoot();
+		j.jenkins
+				.getDescriptorByType(MercurialInstallation.DescriptorImpl.class)
+				.setInstallations(new MercurialInstallation(CACHING_INSTALLATION, "", "hg",
+						false, true, Paths.get("target", "test-data", "custom-cache-dir").toString(), false, "",
```

Comment:
I still need "custom-cache-dir" in the path to be able to assert that the new configuration option was really used as target path for cloning. I'll change it to new File(tmp.newFolder(),"custom-cache-dir").getAbsolutePath() if that's ok for you.